nemo/collections/llm/tools/auto_configurator/runner.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+
+import torch
+from pytorch_lightning.loggers import TensorBoardLogger
+
+from nemo import lightning as nl
+from nemo.collections.common.tokenizers import AutoTokenizer, SentencePieceTokenizer
+from nemo.collections.llm import GPTModel, PreTrainingDataModule
+from nemo.collections.llm.api import pretrain
+from nemo.collections.llm.tools.auto_configurator.core.search_config import search_configs
+from nemo.collections.llm.utils import Config, Partial
+from nemo.lightning.pytorch.optim import CosineAnnealingScheduler, MegatronOptimizerModule
+from nemo.utils import logging
+from nemo.utils.exp_manager import TimingCallback
+
+SUPPORTED_MODELS = [
+    "gpt3",
+    "llama",
+    "mixtral",
+    "mistral",
+    "gemma",
+]
+
+SUPPORTED_TOKENIZERS = [
+    "autotokenizer",
+    "sentencepiece",
+    "huggingface",
+]
+
+
+class AutoConfigurator:
+    """Auto Configurator runner config class."""
+
+    def __init__(
+        self,
+        model_type: str = None,
```

Comment:
Could we take in `Type[ModelConfig]` instead? So you could pass in `Mistral7BConfig`